custom/icds_reports/tasks.py

```diff
@@ -1209,6 +1249,114 @@ def build_incentive_report(agg_date=None):
+# awcagg_row is AggAwc object with 'awc_id', 'is_launched', 'valid_visits', 'expected_visits'
+def get_awc_home_conduct_mismatch_row(performance_row, awcagg_row):
+    # checking if valid_visits or valid_denominator is zero or None as they are treated as valid cases
+    if performance_row[2] in [0, None] or performance_row[3] in [0, None]:
+        home_conduct_from_report = 'None'
+    else:
+        home_conduct_from_report = performance_row[2] / performance_row[3]
+    if awcagg_row[2] in [0, None] or awcagg_row[3] in [0, None]:
+        home_conduct_from_awc = 'None'
+    else:
+        home_conduct_from_awc = awcagg_row[2] / awcagg_row[3]
+    # checking if home conduct (valid_visits/valid_denominator) from incentive report is same as that of aggregate
+    if home_conduct_from_report != home_conduct_from_awc:
+        row_data = [performance_row[0], awcagg_row[0], home_conduct_from_report, home_conduct_from_awc]
+        return row_data
+    return None
+
+
+# performance row is AWCIncentiveReport object with 'awc_id', 'is_launched', 'valid_visits',
+# 'visit_denominator', 'wer_weighed', 'wer_eligible', 'incentive_eligible'
+def get_awc_eligibility_mismatch_row(performance_row):
+    # checking if valid_visits or valid_denominator is zero or None as they are treated as valid cases
+    if performance_row[2] in [0, None] or performance_row[3] in [0, None]:
+        is_eligible = True
+    else:
+        # checking if valid_visits/valid_denominator > 0.6(60%)
+        home_conduct_from_report = performance_row[2] / performance_row[3]
+        if home_conduct_from_report > 0.6:
+            is_eligible = True
+        else:
+            is_eligible = False
+    if is_eligible:
+        # checking if wer_weighed or wer_eligible is zero or None as they are treated as valid cases
+        if performance_row[4] in [0, None] or performance_row[5] in [0, None]:
+            is_eligible = True
+        else:
+            # checking if wer_weighed/wer_eligible > 0.6(60%)
+            weigh_eligibility = performance_row[4] / performance_row[5]
+            if weigh_eligibility > 0.6:
+                is_eligible = True
+            else:
+                is_eligible = False
+    if not performance_row[6]:
+        if is_eligible and performance_row[1]:
+            return [performance_row[0], is_eligible and performance_row[1], performance_row[6]]
+    else:
+        if not (is_eligible and performance_row[1]):
+            return [performance_row[0], is_eligible and performance_row[1], performance_row[6]]
+    return None
+
+
+def awc_is_launched_mismatch_mail_helper(csv_data):
+    csv_columns = ['awc_id_from_performance', 'awc_id_from_aggregate', 'AwwIncentiveReport', 'AggAwc']
+    content = """
+    Please see attached file for is_launched mismatch in awc incentive report
+    """
+    _send_incentive_report_validation_email(content, csv_columns, csv_data)
+
+
+def awc_home_conduct_mismatch_mail_helper(csv_data):
```

Comment:
Why do we need different emails for different checks? Can we send one email showing result of all the checks of performance report? I dont want a check to get missed if one of the email fails to deliver.

Reply:
I thought there will be cases where receipients or content will be different for different cases.If not, we can merge and send them as a single mail